crates/router/src/core/webhooks/outgoing.rs

```diff
@@ -424,46 +430,60 @@ async fn trigger_webhook_to_merchant(
                     )
                     .await?;
                 }
+                Some(updated_event)
             }
         },
-    }
+    };
 
-    Ok(())
+    Ok(updated_event)
 }
 
 fn raise_webhooks_analytics_event(
     state: SessionState,
-    trigger_webhook_result: CustomResult<(), errors::WebhooksFlowError>,
+    trigger_webhook_result: CustomResult<Option<domain::Event>, errors::WebhooksFlowError>,
     content: Option<api::OutgoingWebhookContent>,
     merchant_id: common_utils::id_type::MerchantId,
     event: domain::Event,
 ) {
-    let error = if let Err(error) = trigger_webhook_result {
-        logger::error!(?error, "Failed to send webhook to merchant");
+    let mut event = event;
+    let mut error: Option<serde_json::Value> = None;
 
-        serde_json::to_value(error.current_context())
-            .change_context(errors::ApiErrorResponse::WebhookProcessingFailure)
-            .map_err(|error| {
-                logger::error!(?error, "Failed to serialize outgoing webhook error as JSON");
-                error
-            })
-            .ok()
-    } else {
-        None
-    };
+    match trigger_webhook_result {
+        Ok(updated_event) => {
+            if let Some(updated_event) = updated_event {
+                event = updated_event
+            }
+        }
```

Comment:
Once you update the return type of the function to be `domain::Event` instead of `Option<domain::Event>`, this would have to be suitably updated.